core/src/main/java/com/graphhopper/reader/dem/SRTMGL1Provider.java

```diff
@@ -0,0 +1,155 @@
+        System.out.println(provider.getEle(-22.532854, -65.110474));
+        // 120
+        System.out.println(provider.getEle(38.065392, -87.099609));
+        // 1617
+        System.out.println(provider.getEle(40, -105.2277023));
+        System.out.println(provider.getEle(39.99999999, -105.2277023));
+        System.out.println(provider.getEle(39.9999999, -105.2277023));
+        System.out.println(provider.getEle(39.999999, -105.2277023));
+        // 1015
+        System.out.println(provider.getEle(47.468668, 14.575127));
+        // 1107
+        System.out.println(provider.getEle(47.467753, 14.573911));
+        // 1930
+        System.out.println(provider.getEle(46.468835, 12.578777));
+        // 844
+        System.out.println(provider.getEle(48.469123, 9.576393));
+    }
+
+    @Override
+    byte[] readFile(File file) throws IOException {
+        InputStream is = new FileInputStream(file);
+        BufferedInputStream buff = new BufferedInputStream(is);
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        byte[] buffer = new byte[0xFFFF];
+        int len;
+        while ((len = buff.read(buffer)) > 0) {
+            os.write(buffer, 0, len);
+        }
+        os.flush();
+        Helper.close(buff);
+        return os.toByteArray();
+    }
+
+    int getMinLatForTile(double lat) {
+        return (int) (Math.floor((90 + lat) / latDegree) * latDegree) - 90;
+    }
+
+    int getMinLonForTile(double lon) {
+        return (int) (Math.floor((180 + lon) / lonDegree) * lonDegree) - 180;
+    }
+
+    String getFileName(double lat, double lon) {
+        int lonInt = getMinLonForTile(lon);
+        int latInt = getMinLatForTile(lat);
+        String north = getNorthString(latInt);
+        String east = getEastString(lonInt);
+        String lonString = String.format("%03d", Math.abs(lonInt));
+        return (north + String.format("%02d", Math.abs(latInt)) + east + lonString).toLowerCase();
+    }
+
+    String getDownloadURL(double lat, double lon) {
+        int lonInt = getMinLonForTile(lon);
+        int latInt = getMinLatForTile(lat);
+        String north = getNorthString(latInt);
+        String east = getEastString(lonInt);
+        String lonString = String.format("%03d", Math.abs(lonInt));
+        String dir;
+        if (north.equals("N")) {
+            dir = "North/";
+            if (lat >= 30)
+                dir += "North_30_60/";
+            else
+                dir += "North_0_29/";
+        } else {
+            dir = "South/";
+        }
+
+        return dir + north + String.format("%02d", Math.abs(latInt)) + east + lonString + ".hgt";
+    }
+
+    private String getNorthString(int lat) {
+        if (lat < 0) {
+            return "S";
+        }
+        return "N";
+    }
+
+    private String getEastString(int lon) {
+        if (lon < 0) {
+            return "W";
+        }
+        return "E";
+    }
+
+    @Override
+    public String toString() {
+        return "srtmgl1";
```

Comment:
Is there a reason we use lower case here and upper case in the other? I would prefer having the same case for all even if this means changing the old case.

Reply:
> Is there a reason we use lower case here and upper case in the other?

Oh sorry I missed that. The GMTED provider is also lowercase. I can change both. I just thought it should be the same as the config value.

Reply:
Makes more sense if they have the config value. Would this break some tests or usage scenarios?

Reply:
> Would this break some tests or usage scenarios?

I don't think so. I actually think the toString method is only used to do a System.out:
```
logger.info(this.toString() + " Elevation Provider, from: " + baseUrl + ", to: " + cacheDir + ", as: " + daType);
```